### PR TITLE
Adding features in the search

### PR DESCRIPTION
**Pull request by Frankkwizera**
## Updates:
- added search functionality for searching sites

### CODE REVIEW

File: seshdash/views.py [hunk 1]
```diff
@@ -607,11 +612,25 @@ def silence_alert(request):
...
     latest_bom = BoM_Data_Point.objects.order_by('-time')[0]
-
     latest_bom_data = []
     latest_bom_data.append({"item": "State of Charge", "value":str(latest_bom.soc) + '%' })
     latest_bom_data.append({"item": "Battery Voltage", "value":latest_bom.battery_voltage})
     latest_bom_data.append({"item": "Consumption Data", "value":round(latest_bom.AC_output_absolute, 2)})
     latest_bom_data.append({"item": "Recent Contact", "value": get_timesince(latest_bom.time)})
 
     return HttpResponse(json.dumps(latest_bom_data))
+def search(request):
```
Comment (jlark) [inline seshdash/views.py]: @Frankkwizera please add the `@login_required` decorator here

File: seshdash/views.py [hunk 2]
```diff
@@ -607,11 +612,25 @@ def silence_alert(request):
...
+    data=[]
+    sites = Sesh_Site.objects.all()
```
Comment (jlark) [inline seshdash/views.py]: You don't want to get all the sites. Notice that a user might have permission to view a subset of sites. Make sure you get the users sites

**Comment by jlark** [comment]
@Frankkwizera Thanks for this. Please look at comment inline. Also, please take care of conflicts and recommit.

File: seshdash/views.py (hunk 1 above)
Comment (Frankkwizera) [inline seshdash/views.py]: ok.

File: seshdash/views.py (hunk 2 above)
Comment (Frankkwizera) [inline seshdash/views.py]: i used away of getting all sites.
then suggestion get handled in Javascript after parsing the response.

Comment (jlark) [inline seshdash/views.py]: You should only get the sites the user has permission to view. see function `get_user_sites`

File: seshdash/static/sesh-ui/js/sesh_dashboard.js [hunk 3]
```diff
@@ -239,8 +237,68 @@ get_high_chart( date, HighChartHighPvProduction, HighChartHighCloudCover);
...
+   for (i=0;i < option.length;i++){
+    sitename.push(option[i].value)
+   }
+   for (i=0; i< option.length; i++){
+    siteid.push(option[i].key)
+   }
+   var input = document.getElementById('search');
+   new Awesomplete(input,{list:sitename});
+});
+
```
Comment (jlark) [inline seshdash/static/sesh-ui/js/sesh_dashboard.js]: @Frankkwizera  Please add some comments in this JS
It's very difficult to read as it is.

**Comment by jlark** [comment]
Done

**Comment by jlark** [comment]
merging search

**Pull request closed without merging**